pandas/core/algorithms.py

```diff
@@ -1000,9 +1000,13 @@ def duplicated(
     duplicated : ndarray[bool]
     """
     if hasattr(values, "dtype"):
-        if isinstance(values.dtype, ArrowDtype) and values.dtype.kind in "ifub":
-            values = values._to_masked()  # type: ignore[union-attr]
-
+        if isinstance(values.dtype, ArrowDtype):
+            if values.dtype.kind in "ifub":
+                values = values._to_masked()  # type: ignore[union-attr]
+            else:
+                values = (
+                    values._maybe_convert_datelike_array()  # type: ignore[union-attr]
```

Comment:
let's just make a EA.duplicated for this? xref #27264, #48424

Reply:
updated to implement `ExtensionArray.duplicated`. This is now a bit more general in that it is faster for more than just the pyarrow timestamp and duration types. See updated timings in OP.